### PR TITLE
Webcmdlets set default charset encoding to UTF8

### LINKED ISSUE #10971: Invoke-RestMethod/Invoke-WebRequest should set charset=utf-8 for application/json by default
Recently had an issue trying to use a GitHub API when I put emojis in the POST body:

```powershell
$body = ConvertFrom-Json @{
    tag_name = "tag"
    name = "demo"
    body = "I like 🐛s"
    draft = $true
    prerelease = $true
}

$uri = "https://api.github.com/repos/me/bughunter/releases"
$headers = @{
    GitHubToken = $myToken
    Accept = "application/vnd.github.v3+json"
}

Invoke-RestMethod -Method Post -Uri $uri -Body $body -Headers $headers
```

Sending this published a draft with the description:

```text
I like ??s
```

Took me a while to work out I needed `-ContentType "application/json; charset=utf-8"`, which is fine. Except everything else defaulted (as expected) to UTF-8, so I assumed this would too.

### PR DESCRIPTION
**Pull request by CarloToso**
# PR Summary
Changes the default charset encoding of webcmdlets from Ascii to UTF8
It's just a small fix, it does not solve BOM related problems #11547

## PR Context
Fixes #10971

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/main/reference/7.3/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [ ] Not Applicable
    - **OR**
    - [x] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [x] Issue filed: https://github.com/MicrosoftDocs/PowerShell-Docs/issues/9394
- **Testing - New and feature**
    - [ ] N/A or can only be tested interactively
    - **OR**
    - [x] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [ ] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and opened an issue in the relevant tool repository. This may include:
        - [ ] Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode
        (which runs in a different PS Host).
            - [ ] Issue filed: 
        - [ ] Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
            - [ ] Issue filed: 
        - [ ] Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
            - [ ] Issue filed: 
        - [ ] Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).
            - [ ] Issue filed:

### CODE REVIEW

**Review by SteveL-MSFT — APPROVED** [review]

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebRequestPSCmdlet.Common.cs [hunk 1]
```diff
@@ -1701,7 +1701,7 @@ internal long SetRequestContent(HttpRequestMessage request, string content)
             if (content == null)
                 return 0;
 
-            Encoding encoding = null;
+            Encoding encoding = Encoding.UTF8;
```
Comment (iSazonov) [inline src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebRequestPSCmdlet.Common.cs:1704]: Is this right place for the change?
We have 
```c#
    internal static class ContentHelper
    {
        #region Constants

        // default codepage encoding for web content.  See RFC 2616.
        private const string _defaultCodePage = "ISO-8859-1";
```

**Review by iSazonov — CHANGES_REQUESTED** [review]

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebRequestPSCmdlet.Common.cs (hunk 1 above)
Comment (CarloToso) [reply src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebRequestPSCmdlet.Common.cs:1704]: I checked and apparently most of the methods in the ContentHelper class are never used, so modifiying _defaultCodePage does not work. The ContentHelper class might need a rewrite, but I think it's beyond the scope of this small fix,

Comment (iSazonov) [reply src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebRequestPSCmdlet.Common.cs:1704]: The `encoding` is used below in `StreamHelper.EncodeToBytes()`, and if it is null the `encoding` defaults to `_defaultCodePage `.

Comment (CarloToso) [reply src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebRequestPSCmdlet.Common.cs]: I fixed it, following your suggestion

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/ContentHelper.Common.cs [hunk 2]
```diff
@@ -40,13 +39,6 @@ internal static Encoding GetDefaultEncoding()
             return GetEncodingOrDefault((string)null);
         }
 
-        internal static Encoding GetEncoding(HttpResponseMessage response)
-        {
-            // ContentType may not exist in response header.
-            string charSet = response.Content.Headers.ContentType?.CharSet;
-            return GetEncodingOrDefault(charSet);
-        }
-
```
Comment (iSazonov) [inline src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/ContentHelper.Common.cs:49]: Please revert the unrelated change.

Comment (CarloToso) [reply src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/ContentHelper.Common.cs]: I reverted it

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/ContentHelper.Common.cs [hunk 3]
```diff
@@ -39,14 +38,14 @@ internal static Encoding GetDefaultEncoding()
         {
             return GetEncodingOrDefault((string)null);
         }
-
+
```
Comment (iSazonov) [inline src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/ContentHelper.Common.cs:41]: Please revert.

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/ContentHelper.Common.cs [hunk 4]
```diff
@@ -39,14 +38,14 @@ internal static Encoding GetDefaultEncoding()
...
-
+        
         internal static Encoding GetEncoding(HttpResponseMessage response)
         {
             // ContentType may not exist in response header.
             string charSet = response.Content.Headers.ContentType?.CharSet;
             return GetEncodingOrDefault(charSet);
         }
-
+
```
Comment (iSazonov) [inline src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/ContentHelper.Common.cs:48]: The same.

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/ContentHelper.Common.cs (hunk 3 above)
Comment (CarloToso) [reply src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/ContentHelper.Common.cs]: Done

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/ContentHelper.Common.cs (hunk 4 above)
Comment (CarloToso) [reply src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/ContentHelper.Common.cs]: Done

**Comment by iSazonov** [comment]
@SteveL-MSFT I think the PR makes sense but we need WG/Committee conclusion since we are moving away from RFC standard.

**Comment by CarloToso** [comment]
@SteveL-MSFT do you have any updates?

**Comment by CarloToso** [comment]
> @CarloToso please read the following Contributor License Agreement(CLA). If you agree with the CLA, please reply with the following information.
> 
> ```
> @microsoft-github-policy-service agree [company="{your company}"]
> ```
> 
> > Options:
> > 
> > * (default - no company specified) I have sole ownership of intellectual property rights to my Submissions and I am not making Submissions in the course of work for my employer.
> > * (when company given) I am making Submissions in the course of work for my employer (or my employer has intellectual property rights in my Submissions by contract or applicable law). I have permission from my employer to make Submissions and enter into this Agreement on behalf of my employer. By signing below, the defined term “You” includes me and my employer.
> 
> Contributor License Agreement

@microsoft-github-policy-service agree [company="{}"]

**Comment by iSazonov** [comment]
@CarloToso You should sign CLA before we can merge.

**Comment by CarloToso** [comment]
> @CarloToso the command you issued was incorrect. Please try again.
> 
> Examples are:
> 
> ```
> @microsoft-github-policy-service agree
> ```
> 
> and
> 
> ```
> @microsoft-github-policy-service agree company="your company"
> ```

@microsoft-github-policy-service agree

**Comment by CarloToso** [comment]
@iSazonov I fixed the tests, an error remains, but I don't think it's related to this pull request

**Comment by iSazonov** [comment]
@CarloToso For CLA please one command without comments.

**Comment by CarloToso** [comment]
@microsoft-github-policy-service agree

**Comment by CarloToso** [comment]
@microsoft-github-policy-service agree

**Comment by CarloToso** [comment]
It seems the cla-bot is broken (https://cla.microsoft.com/ gives 403 error)

**Review by iSazonov — APPROVED** [review]

**Comment by CarloToso** [comment]
@microsoft-github-policy-service agree

**Comment by iSazonov** [comment]
@SteveL-MSFT Please update you review.

**Comment by daxian-dbw** [comment]
Marking Cmdlet WG to review.

**Comment by SteveL-MSFT** [comment]
Added to WG queue

**Comment by SteveL-MSFT** [comment]
@PowerShell/wg-powershell-cmdlets reviewed this and compared the behavior to curl which correctly handles the UTF-8 BOM so we agree that the default should be UTF-8 instead of Ascii"

**Review by SteveL-MSFT — APPROVED** [review]

**Comment by iSazonov** [comment]
@CarloToso Thanks for your contribution!

**Comment by CarloToso** [comment]
@iSazonov Thank you for your guidance
